api/src/aggregate/aggregate.service.ts

```diff
@@ -73,4 +83,30 @@ export class AggregateService {
       return null
     }
   }
+
+  async findLatestByAggregatorId(latestAggregateByIdDto: LatestAggregateByIdDto) {
+    const { aggregatorId } = latestAggregateByIdDto
+
+    const redisKey = `latestAggregate:${aggregatorId.toString()}`
+    const rawResult = await this.redis.get(redisKey)
+
+    if (!rawResult) {
+      return await this.findLatestByAggregatorIdFromPrisma(aggregatorId)
+    }
+    const { timestamp, value } = JSON.parse(rawResult)
+    return { timestamp, value: BigInt(value) }
+  }
+
+  async findLatestByAggregatorIdFromPrisma(aggregatorId) {
+    const prismaResult = await this.prisma.aggregate.findFirst({
+      where: { aggregatorId },
+      orderBy: { timestamp: 'desc' }
+    })
+    if (!prismaResult) {
+      return null
```

Comment:
How about keeping the response format?

```typescript
return {timestamp, value: null}
```

Reply:
set to return
`return {timestamp: "", value: null}`